service/gcs/core/gcs/gcs.go

```diff
@@ -105,6 +107,20 @@ func (e *containerCacheEntry) RemoveMappedVirtualDisk(disk prot.MappedVirtualDis
 	delete(e.MappedVirtualDisks, disk.Lun)
 	return nil
 }
+func (e *containerCacheEntry) AddMappedDirectory(dir prot.MappedDirectory) error {
+	if _, ok := e.MappedDirectories[dir.Port]; ok {
+		return errors.Errorf("a mapped directory with port %d is already attached to container %s", dir.Port, e.ID)
+	}
+	e.MappedDirectories[dir.Port] = dir
+	return nil
+}
+func (e *containerCacheEntry) RemoveMappedDirectory(dir prot.MappedDirectory) error {
```

Comment:
That's fair. That also maintains parity with the unmount code, which only actually performs the unmount if the directory exists and is actually mounted. I can just change this to log an error.